pkg/adapter/adapter.go

```diff
@@ -154,14 +155,18 @@ func (a *Adapter) PollForMessages(stopCh <-chan struct{}) error {
 
 	retryChan := make(chan bool)
 	defer a.rmqHelper.CleanupRabbitMQ(a.connection, logger)
+	listenToConnections := true
 	for {
-		a.connection, a.channel, err = a.rmqHelper.SetupRabbitMQ(rabbit.VHostHandler(a.config.RabbitURL, a.config.Vhost), rabbit.ChannelQoS, logger)
+		a.connection, a.channel, err = a.rmqHelper.SetupRabbitMQ(rabbit.VHostHandler(a.config.RabbitURL, a.config.Vhost), rabbit.ChannelQoS, logger, listenToConnections)
 		if err != nil {
 			logger.Errorf("error creating RabbitMQ connections: %s, waiting for a retry", err)
 		} else {
 			queue, err = a.channel.QueueInspect(a.config.QueueName)
 			if err != nil {
+				listenToConnections = false
```

Comment:
What's the point of setting the listenToConnections to false here if we set it back to true on L182 before the next call to SetupRabbitMQ

Reply:
Because there is a continue after setting it to false, so in the case of a fail with queue inspect it gets called with false